File: src/main/java/com/wikia/tibia/objects/Mount.java
package com.wikia.tibia.objects;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.wikia.tibia.enums.Status;
import com.wikia.tibia.enums.YesNo;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Mount extends WikiObject {

    private Integer speed;
    private String tamingMethod;
    private YesNo bought;
    private Integer price; // unit is Tibia Coins
    private String achievement; // this could link to Achievement
    private Integer lightradius;
    private Integer lightcolor;
    private String artwork;

    @Builder
    private Mount(String name, String implemented, String notes, String history, Status status, Integer speed,
                  String tamingMethod, YesNo bought, Integer price, String achievement, Integer lightradius,
                  Integer lightcolor, String artwork) {
        super(name, null, null, null, implemented, notes, history, status);
        this.speed = speed;
        this.tamingMethod = tamingMethod;
        this.bought = bought;
        this.price = price;
        this.achievement = achievement;
        this.lightradius = lightradius;
        this.lightcolor = lightcolor;
        this.artwork = artwork;
    }

    @JsonGetter("taming_method")
    private String getTamingMethod() {
        return tamingMethod;
    }

    @Override
    public void setDefaultValues() {
        // TODO implement this method
    }
}

File: src/main/kotlin/com/wikia/tibia/objects/Mount.kt
package com.wikia.tibia.objects

import com.fasterxml.jackson.annotation.JsonGetter
import com.wikia.tibia.enums.Status
import com.wikia.tibia.enums.YesNo
import com.wikia.tibia.utils.ObjectUtils.isEmpty

data class Mount(
  val name: String,
  val implemented: String,
  val notes: String,
  val history: String,
  val status: Status,
  val speed: Int,
  @get:JsonGetter("taming_method") val tamingMethod: String,
  val bought: YesNo,
  val tournament: YesNo, // unit is Tibia Coins
  val price: Int, // this could link to Achievement
  val achievement: String,
  val lightradius: Int,
  val lightcolor: Int,
  val artwork: String
) : WikiObject() {

  override fun setDefaultValues() {
    if (isEmpty(implemented)) {
      this.copy(implemented = "?")
    }
  }
}
